src/GslCore/GslcProcess.fs

```diff
@@ -129,12 +129,51 @@ let cleanLongSlicesInPartsList (p:pragmaTypes.PragmaCollection) (l:DNASlice list
                                     // exn if necessary
                                     failwithf "%s" (String.Join(";",messages))
             }
+        else if (s.pragmas.ContainsKey("amp") && s.pragmas.ContainsKey("inline"))
+        then
+            failwithf "ERROR: Part %s has incompatible pragmas #amp and #inline. Please pick one." s.sliceName
         else s)
 
 /// Promote long slices to regular rabits to avoid trying to build
 /// impossibly long things with oligos.
 let cleanLongSlices _ (a:DnaAssembly) =
     ok {a with dnaParts = cleanLongSlicesInPartsList a.pragmas a.dnaParts}
+    
+/// Demote short slices to inlinest to build in oligos
+/// rather than PCR 
+let cleanShortSlicesInPartsList (p:pragmaTypes.PragmaCollection) (l:DNASlice list) =
+    l |> List.map (fun s ->
+        if (s.sliceType = REGULAR &&
+            (s.pragmas.ContainsKey("inline") || (s.dna.Length <= 30)) &&
+            not (s.pragmas.ContainsKey("amp")))
+        then
+            {s with
+                sliceType = INLINEST;
+                dnaSource =
+                    match s.pragmas.TryGetOne("dnasrc") with
+                    | Some(x) -> x
+                    | None ->
+                        match p.TryGetOne("refgenome") with
+                        | None -> "synthetic"
+                        | Some(x) -> x
+                // add in an amp tag on this guy too, since we are now comitting to

```

Comment:
comment is wrong now.... (on my first summer job, I still remember someone telling me - "debug only code, not comments, they can be lies"

Reply:
Good catch, edited the comments :)